soh/soh/Enhancements/speechsynthesizer/SpeechSynthesizer.h

```diff
@@ -0,0 +1,38 @@
+//
+//  SpeechSynthesizer.h
+//  libultraship
+//
+//  Created by David Chavez on 22.11.22.
+//
+
+#ifndef SOHSpeechSynthesizer_h
+#define SOHSpeechSynthesizer_h
+
+#include <stdio.h>
+
+class SpeechSynthesizer {
+  public:
+    static SpeechSynthesizer* Instance;
+    SpeechSynthesizer();
+
+    bool Init(void);
+    void Uninitialize(void);
+    virtual void Speak(const char* text, const char* language) = 0;
```

Comment:
what's the logic behind having this be a char* instead of taking in a std::string here?

Reply:
that seems reasonable, looking at it from a purely SoH perspective it feels odd having a bunch of `.c_str()` calls everywhere. glad to have this decision documented on this PR